### ISSUE TITLE
Containers Removed by tsdproxy Remain Marked as ‘Connected’ in Tailscale Dashboard

### ISSUE THREAD

**Issue by jasonc** [almeidapaulopt/tsdproxy#128]
**Describe the bug**
When a container is removed from the Docker host and `tsdproxy` removes the proxy for the container, the container still appears as "Connected" (with a green dot beside it) in the "Last Seen" column of the Tailscale dashboard. This status persists for over 12 hours, even after the container is no longer present. In previous versions (e.g., `tsdproxy` version 1.1.X), the container's status in the Tailscale dashboard would change to a date (with a grey dot beside it) after disconnection, allowing ephemeral hosts to be cleaned up automatically.

**Expected behavior**
When a container is removed from the Docker host and `tsdproxy` removes its proxy, the container's status in the Tailscale dashboard should transition from "Connected" to a date (with a grey dot beside it). This would enable Tailscale to recognize that the container is no longer active and clean up ephemeral hosts accordingly.

**Send config**
Here is the `tsdproxy.yaml` configuration file:
```
defaultproxyprovider: default
docker:
    local:
        host: unix:///var/run/docker.sock
        targethostname: 172.31.0.1
files: {}
tailscale:
    providers:
        default:
            authkey: tskey-auth-PRIVATE
            controlurl: https://controlplane.tailscale.com
    datadir: /data/
http:
    hostname: 0.0.0.0
    port: 8080
log:
    level: debug
    json: false
proxyaccesslog: true
```

**Logs**
```
2:53PM INF Container fd12d28cf9a21229d10d4386c23e4c2fd78efb982e66e9c4445b4ad7403382f1 started docker=local module=proxymanager
2:53PM DBG Adding target module=proxymanager targetID=fd12d28cf9a21229d10d4386c23e4c2fd78efb982e66e9c4445b4ad7403382f1
2:53PM INF Trying to auto detect target URL container=/sample-nginx docker=local module=proxymanager try=0
2:53PM DBG trying to connect to internal port container=/sample-nginx docker=local hostname=172.31.0.1 module=proxymanager port=80
2:53PM DBG Failed to connect address=172.17.0.2 container=/sample-nginx docker=local module=proxymanager port=80
2:53PM DBG Error connecting to internal port error="no valid target found for internal ports " container=/sample-nginx docker=local module=proxymanager
2:53PM INF Successfully connected using docker network gateway and exposed port address=172.17.0.1 container=/sample-nginx docker=local module=proxymanager port=8111
2:53PM DBG Creating proxy module=proxymanager proxy=sample-nginx
2:53PM INF setting up proxy hostname=sample-nginx module=proxymanager
2:53PM DBG initializing proxy hostname=sample-nginx module=proxymanager proxyURL=https://sample-nginx targetURL=http://172.17.0.1:8111
2:53PM DBG Setting up tailscale server hostname=sample-nginx module=proxymanager tailscale=default
2:53PM DBG Proxy server created successfully hostname=sample-nginx module=proxymanager proxyname=sample-nginx
2:53PM INF starting proxy module=proxymanager name=sample-nginx proxyname=sample-nginx
2:53PM INF tsnet running state path /data/default/sample-nginx/tailscaled.state module=proxymanager tailscale=default
2:53PM INF tsnet starting with hostname "sample-nginx", varRoot "/data/default/sample-nginx" module=proxymanager tailscale=default
2:53PM INF LocalBackend state is NeedsLogin; running StartLoginInteractive... module=proxymanager tailscale=default
2:53PM INF Generating TLS certificate Hostname=sample-nginx module=proxymanager tailscale=default
2:53PM INF AuthLoop: state is Running; done module=proxymanager tailscale=default
2:53PM INF TLS certificate generated Hostname=sample-nginx module=proxymanager tailscale=default
2:53PM INF Container fd12d28cf9a21229d10d4386c23e4c2fd78efb982e66e9c4445b4ad7403382f1 stopped docker=local module=proxymanager
2:53PM DBG Stopping target module=proxymanager targetID=fd12d28cf9a21229d10d4386c23e4c2fd78efb982e66e9c4445b4ad7403382f1
2:53PM INF Removed proxy for container module=proxymanager proxy=sample-nginx
```

**Additional context**
This behavior differs from what was observed in previous versions (e.g., tsdproxy version 1.1.X), where containers would be properly marked as disconnected in the Tailscale dashboard. This change impacts the management of ephemeral hosts, as they are no longer automatically deleted by Tailscale.

**Comment by baumheld** [comment]
Some thoughts

1. Did you enable `Ephemeral` when you created your auth key in the tailscale settings page? 
![image](https://github.com/user-attachments/assets/cb2b775f-311a-4720-8a30-a16647d0b737)
I noticed that when I use an auth key without Ephemeral enabled, setting `tsdproxy.ephemeral: true` for TSDProxy in Docker Compose doesn't have any effect. The machines don't get the tag "ephemeral" in Tailscale admin console and therfore stay forever and have to be removed manually


2. The docs say you can use `-state=mem:`
     > Another way is if you created an ephemeral device by running tailscaled with the --state=mem: flag. If the device is running   Tailscale v1.30 or later, when the device exits, the tailscaled daemon itself performs tailscale logout which immediately removes the device from your tailnet. [source](https://tailscale.com/kb/1111/ephemeral-nodes)

    But I guess this setting has to be implemented first in TSDProxy. Or is there a way to use this in TSDProxy?

**Comment by jasonc** [comment]
> 1. Did you enable `Ephemeral` when you created your auth key in the tailscale settings page?

I tested with a key that had Ephemeral enabled and with a key with Ephemeral not enabled. In both cases, after the container has been destroyed and tsdproxy logs "Removed proxy for container", the container (machine in Tailscale speak) remains connected in the [machines area of the Tailscale dashboard](https://login.tailscale.com/admin/machines).

I see "Connected," but when using a non-ephemeral key I would expect to see a date. Here's a screenshot of a destroyed container (added by tsdproxy initially) and another unrelated device that is disconnected. (I would expect the destroyed container to also show as disconnected, but it doesn't.)

<img width="197" alt="Screenshot 2024-12-28 at 9 38 45 AM" src="https://github.com/user-attachments/assets/b5f3b6b5-33e3-46ae-8f0d-5310695e3f6a" />

In the case of using an Ephemeral key, I would expect to see a disconnected date for a short period of time and eventual the device to go away from the Tailscale dashboard. But again, it never gets to the disconnected stated; it always shows connected.

**Comment by almeidapaulopt** [comment]
fixed in v1.3.1 https://github.com/almeidapaulopt/tsdproxy/commit/5716079b448b688042535c853351e8489a93f931
